wordpress-mobile/WordPress-iOS
# Dashboard: Unified Menu

Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/21492

I kept the QuickActions cell but changed the layout to vertical and updated the cell design. The rest of the code is largely the same. It was also able to remove four iOS 15 deprecations, which is nice. And the dashboard is now vastly simplified.

I also added a way to remove Quick Start tour for a current blog.

### Not in the Scope

- Update the blog details page (more menu)
- Adding details labels to the new menus items (it will require more work and will be done separately)

<img width="453" alt="Screenshot 2023-09-12 at 10 45 19 AM" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/d65bc89e-ca11-44b9-84a4-19c6c82b4c47">

## To test:

- Verify that the cell is highlighted as you select it and is deselected when you return to the dashboard: [recording](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/a5e31da1-9a97-4eea-9e74-928c20a9a20a)

### Site with no Jetpack

- Add a self-hosted site without Jetpack and verify that the app shows a "more" menu instead of the dashboard
- Select a WP.COM site from a site picker and verify that it switched to dashboard

### Quick Start

- Verify that the app leads you through the entire flow
- Verify that the new unified menu has spotlights when needed: [recording](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/ba3c3b1f-01cc-48d5-bf21-57363294f29c)
- Verify that tours that require you to open blog details prompt you to tap "More": [recoding](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/55706b52-71bc-45ef-a4a6-db6492a918b3)


### WordPress
- Verify that WordPress is unaffected

## Regression Notes
1. Potential unintended areas of impact: Dashboard and Menu
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual tests
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:


UI Changes testing checklist:
